pennylane/gradients/parameter_shift.py

```diff
@@ -757,7 +757,9 @@ def _get_var_with_second_order(pdA2, f0, pdA):
             and the gradient recipe contains an unshifted term, this value is used,
             saving a quantum evaluation.
     """
-    return qml.math.array(pdA2 - 2 * f0 * pdA)
+    if any(isinstance(var, np.ndarray) for var in [pdA2, f0, pdA]):

```

Comment:
Necessary because qml.math.array breaks differentiability for Torch tensor.

Reply:
Worth adding a comment on this.

Reply:
Added!